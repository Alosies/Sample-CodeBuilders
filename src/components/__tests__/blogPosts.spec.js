import { shallowMount, createLocalVue } from '@vue/test-utils'
import BlogPost from '@/components/BlogPosts'
import Vuetify from 'vuetify'
import Vue from 'vue'
import Vuex from 'vuex'

Vue.use(Vuetify)

const localVue = createLocalVue()
localVue.use(Vuex)

const store = new Vuex.Store({
  state: {
    allPosts: []
  },
  actions: {
    fetchPosts() {
      return []
    }
  },
  getters: {
    allPosts() {
      return []
    }
  }
})

describe('HelloWorld.vue', () => {
  it('BlogPost Component renders the posts', () => {
    const wrapper = shallowMount(BlogPost, {
      store,
      localVue
    })
    expect(wrapper.html()).toMatchSnapshot()
  })
})
