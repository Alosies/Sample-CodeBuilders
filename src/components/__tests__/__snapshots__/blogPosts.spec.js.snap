// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HelloWorld.vue BlogPost Component renders the posts 1`] = `
<div>
  <div class="heading">
    <v-icon-stub size="3rem">
      ballot
    </v-icon-stub>
    <h1 class="primary--text">
      Blog Posts
    </h1>
  </div>
  <div class="postsContainer"></div>
</div>
`;
